api/src/main/java/jakarta/servlet/http/HttpServlet.java

```diff
@@ -501,6 +509,10 @@ protected void doTrace(HttpServletRequest req, HttpServletResponse resp) throws
         out.print(buffer.toString());
     }
 
+    private boolean isSensitiveHeader(String headerName) {
+        return HEADER_SENSITIVE.contains(headerName);
+    }
+
```

Comment:
Does this really need a separate method?

Reply:
In this [RFC](https://www.rfc-editor.org/rfc/rfc9110.html#name-trace) don't have the list of sensitive headers. Then include on this method because Java method will be seen by JIT to compiler separated of #doTrace and will be better performance.

Reply:
All JIT is likely to do is in-line the method - which is what I am suggesting you do anyway. It isn't as if the separate method adds to the readability of the code.

Reply:
Either in line it, or make it protected so that an extender can change what headers are considered safe.